index.js

```diff
@@ -1133,6 +1136,8 @@
   //.
   //. The Either type satisfies the [Semigroup][], [Monad][], and [Extend][]
   //. specifications.
+  //.
+  //. This type is exported as `S.EitherType` for use with [sanctuary-def][].
```

Comment:
This deserves a separate section in the documentation. We can then have `Either :: TypeRep Either` and `EitherType :: Type -> Type -> Type`.